src/HasMedia/HasMediaTrait.php

```diff
@@ -282,31 +298,14 @@ public function addMediaConversion($name)
     }
 
     /**
-     * Delete the model. The extra logic isn't handled in a model event since the boot function is unreliable
-     * for currently unknown reasons.
-     *
-     * @return bool
-     */
-    public function delete()
-    {
-        if (!parent::delete()) {
-            return false;
-        }
-
-        $this->media()->get()->map(function ($media) {
-            $media->delete();
-        });
-
-        return true;
-    }
-
-    /**
      * Delete the model, but preserve all the associated media.
      *
      * @return bool
      */
     public function deletePreservingMedia()
     {
+        $this->deletePreservingMedia = true;
```

Comment:
In edge cases where `delete` would return false, you'd now have a model with `deletePreservingMedia` true, even if you set it false before. Feels slightly off to me, but I don't know if I prefer this or added complexity in this method by keeping the old value. Pinging @freekmurze for this one.